### PR TITLE
Require __BYTE_ORDER__ to be defined for bi-endian target architectures.

### PR DESCRIPTION
**Pull request by briansmith**

### CODE REVIEW

**Comment by briansmith** [comment]
cc @uweigand @erichte-ibm @pkubaj. The basic idea here is to ensure that __BYTE_ORDER__ is defined properly since @uweigand's contribution to crypto/internal.h requires __BYTE_ORDER__ to be defined correctly, and to avoid using any other indication of (big-)endianness.

**Review by ecnelises — COMMENTED** [review]
maybe off-topic but isn't `mipsel` little-endian?

File: include/ring-core/target.h [hunk 1]
```diff
@@ -45,16 +58,11 @@
...
-#define OPENSSL_64_BIT
-#define OPENSSL_RISCV64
 #elif defined(__s390x__)
 #define OPENSSL_64_BIT
 #define OPENSSL_S390X
-#elif defined(__wasm__)
-#define OPENSSL_32_BIT
 #else
 // Note BoringSSL only supports standard 32-bit and 64-bit two's-complement,
 // little-endian architectures. Functions will not produce the correct answer
```
Comment (DavidHorton5339) [inline include/ring-core/target.h:68]: Please update comment to reflect that some big-endian targets are supported.

**Review by DavidHorton5339 — APPROVED** [review]
LGTM - though please see suggestion about comment.

**Comment by DavidHorton5339** [comment]
> maybe off-topic but isn't `mipsel` little-endian?

MIPS is bi-endian, so although mipsel is the little-endian variant, OPENSSL_MIPS is insufficient on it own to uniquely identify the target elsewhere in the code. That's why __BYTE_ORDER__ is needed too, and the purpose of this PR is to make sure that is set for bi-endian target architectures where this uncertainty exists.

Perhaps it would be clearer to prepend 'Require __BYTE_ORDER__ to be defined for bi-endian target architectures.' to the comment block starting on line 45, and to replace 'big-endianness' with 'endianness'.

File: include/ring-core/target.h (hunk 1 above)
Comment (briansmith) [reply include/ring-core/target.h]: I just removed the comment.

**Comment by briansmith** [comment]
> replace 'big-endianness' with 'endianness'.

I implemented this suggestion.

**Comment by briansmith** [comment]
Thanks for reviewing this. I really appreciate it.

**Pull request merged**